data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -136,66 +146,124 @@ private void parseFile(String filePath){
+        targetValue.ifPresent(targetValues::add);
     }
 
-    public boolean matchesIndividualEntry(Record<Event> record, String matchKey){
-        if(individualMappings.containsKey(matchKey)){
-            record.getData().put(translateProcessorConfig.getTarget(), individualMappings.get(matchKey));
-            return true;
+    private Optional<String> matchesIndividualEntry(final String sourceKey) {
+        if (individualMappings.containsKey(sourceKey)) {
+            return Optional.of(individualMappings.get(sourceKey));
         }
-        return false;
+        return Optional.empty();
     }
 
-    public boolean matchesRangeEntry(Record<Event> record, String matchKey){
-        if(!NumberUtils.isParsable(matchKey)){
-            return false;
+    private Optional<String> matchesRangeEntry(final String sourceKey) {
+        if (!NumberUtils.isParsable(sourceKey)) {
+            return Optional.empty();
         }
-        Float floatKey = Float.parseFloat(matchKey);
-        for(Map.Entry<Range<Float>, String> rangeEntry : rangeMappings.entrySet()) {
+        Float floatKey = Float.parseFloat(sourceKey);
+        for (Map.Entry<Range<Float>, String> rangeEntry : rangeMappings.entrySet()) {
             Range<Float> range = rangeEntry.getKey();
             if (range.contains(floatKey)) {
-                record.getData().put(translateProcessorConfig.getTarget(), rangeEntry.getValue());
-                return true;
+                return Optional.of(rangeEntry.getValue());
             }
         }
-        return false;
+        return Optional.empty();
     }
 
-    public boolean matchesPatternEntry(Record<Event> record, String matchKey){
-        //todo
-        if(!Objects.nonNull(patternMappings)){
-            return false;
+    private Optional<String> matchesPatternEntry(final String sourceKey) {
+        if (!compiledPatterns.isEmpty()) {
+            for (Pattern pattern : compiledPatterns.keySet()) {
+                if (pattern.matcher(sourceKey).matches()) {
+                    return Optional.of(compiledPatterns.get(pattern));
+                }
+            }
+            if (!translateProcessorConfig.getRegexParameterConfiguration().getExact()) {
+                for (String pattern : patternMappings.keySet()) {
+                    if (pattern.contains(sourceKey)) {
+                        return Optional.of(patternMappings.get(pattern));
+                    }
+                }
+            }
         }
-        for(String pattern : patternMappings.keySet()){
-            if(Pattern.matches(pattern, matchKey)){
-                record.getData().put(translateProcessorConfig.getTarget(), patternMappings.get(pattern));
-                return true;
+        return Optional.empty();
+    }
+
+    private void addTargetToRecords(Object sourceObject, List<String> targetValues, Object recordObject) {
+        String targetField = translateProcessorConfig.getTarget();
+        if (!targetValues.isEmpty()) {
```

Comment:
Nit. Better way to write is 
```
if (targetValues.isEmpty()) {
    return;
}
```
This way the rest of the code doesn't have to be indented in.